### ISSUE TITLE
Column number should start at 1 and not 0

### ISSUE THREAD

**Issue by GodTamIt** [rust-lang/annotate-snippets-rs#17]
Given this code:

```
let snippet = Snippet {
    title: Some("failed to parse file".to_owned()),
    footer: vec![],
    slices: vec![Slice {
        source: r#"This is an example
content of the slice
which will be annotated
with the list of annotations below."#
            .to_string(),
        line_start: 26,
        origin: Some("chris.beep".to_owned()),
        fold: false,
        annotations: vec![SourceAnnotation {
            label: "hello world".to_owned(),
            annotation_type: AnnotationType::Error,
            range: (13, 20),
        }],
    }],
};
```

This results in the following output:
```
error: failed to parse file
  --> chris.beep:26:13
   |
26 | This is an example
   |              ^^^^^^^ hello world
27 | content of the slice
28 | which will be annotated
29 | with the list of annotations below.
```

Because the output should be for humans, shouldn't the location be `chris.beep:26:14`?

**Comment by digama0** [comment]
I just run across this issue as well. It also affects editors like VS Code that parse console output for expressions of the form `foo.txt:1:2` and turn them into links into the file `foo.txt`, line 1, column 2 (1-based).

I think the fix is the following (in [from_snippet.rs:95](https://github.com/rust-lang/annotate-snippets-rs/blob/master/src/display_list/from_snippet.rs#L95)) but I don't know what needs fixing for tests and such:
```diff
        let mut col = 1;
        let mut row = slice.line_start;

        for item in body.iter() {
            if let DisplayLine::Source {
                line: DisplaySourceLine::Content { range, .. },
                ..
            } = item
            {
                if annotation.range.0 >= range.0 && annotation.range.0 <= range.1 {
                    println!("hi");
-                   col = annotation.range.0 - range.0;
+                   col = annotation.range.0 - range.0 + 1;
                    break;
                }
                row += 1;
            }
        }

**Comment by GodTamIt** [comment]
Should be fixed with #22 landing now. Closing this.
